### PR TITLE
Update to h5pp/1.11.1

### PR DESCRIPTION
**Pull request by DavidAce**
**h5pp/1.11.1**

Bump version

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by DavidAce** [comment]
The build fails because hdf5/1.14.13 (the latest) is missing binaries for armv8. These binaries exist for hdf5/1.14.2 though.

I'm not sure how to resolve this type of issue. Any help is appreciated.

**Comment by uilianries** [comment]
I just started an internal CI build to generate the missing hdf5/1.14.3 package. Sorry the inconvenience.

**Review by franramirez688 — DISMISSED** [review]

**Review by uilianries — CHANGES_REQUESTED** [review]
avoid ranged versions for any packages. It will result in missing dependency in future PRs.

**Review by franramirez688 — COMMENTED** [review]
Absolutely @uilianries 
I approved it, but I didn't realize about those version ranges

File: recipes/h5pp/all/conanfile.py [hunk 1]
```diff
@@ -62,13 +64,16 @@ def config_options(self):
             del self.options.with_zlib
         else:
             self.options["hdf5"].with_zlib = self.options.with_zlib
+        if Version(self.version) < "1.11.1" or self.settings.compiler != "gcc":
+            # h5pp only supports quadmath with GNU compilers
+            del self.options.with_quadmath
 
     def requirements(self):
-        self.requires("hdf5/1.14.0", transitive_headers=True, transitive_libs=True)
+        self.requires("hdf5/[>=1.10.0 <1.15]", transitive_headers=True, transitive_libs=True)
```
[the remark this answers is not in the document]
Comment (DavidAce) [reply recipes/h5pp/all/conanfile.py]: Fixed! Thanks!

**Review by uilianries — APPROVED** [review]
LGTM

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**